pandas/core/sorting.py

```diff
@@ -436,7 +436,13 @@ def nargsort(
     if not ascending:
         non_nans = non_nans[::-1]
         non_nan_idx = non_nan_idx[::-1]
-    indexer = non_nan_idx[non_nans.argsort(kind=kind)]
+    # GH#54072
+    # argsort does not support mixed int/string Index
+    try:
+        indexer = non_nan_idx[non_nans.argsort(kind=kind)]
+    except TypeError as err:
+        msg = "'<' not supported between "
```

Comment:
this looks like the message just cuts off?  what is the message in err?

Reply:
Thank you for the comments.
Yes, I cut the message off because in tests I had both: "'<' not supported between instances of 'int' and 'str'" and "'<' not supported between instances of 'str' and 'int'" and sometimes messgaes did not match. 
I corrected error message in the function `nargsort` and tests, now all error messages match.